elasticsearch/tests/test_elasticsearch.py

```diff
@@ -8,6 +8,8 @@
 # The versions below were the current supported versions at time of writing (2022-08-11)
 @pytest.mark.parametrize('version', ['6.8.23', '7.17.5', '8.3.3'])
 def test_docker_run_elasticsearch(version):
-    with ElasticSearchContainer(f'elasticsearch:{version}') as es:
+    # adding the mem_limit as per: https://stackoverflow.com/a/75701019
+    # could also add (but not necessary): .with_env('discovery.type', 'single-node')
+    with ElasticSearchContainer(f'elasticsearch:{version}', mem_limit='3G') as es:
```

Comment:
In tc-java, we configure the Elastic process using 
```
-Xms2147483648
-Xmx2147483648
```
don't we need to do the same here?

Reply:
hm, i can update this to 2g to make it match

Reply:
oh, so - it depends on how much faith we have in the commitment to backwards compat on the side of the elastic team

in the logs, the container prints a 1.5g heap on startup. do we think that will change? otherwise, i guess i have to explicitly set the heap, so that we know the mem_limit will be high enough in the future

Reply:
should i implement `with_copy_file_to_container` in this same pr? it is missing for parity with tc-java

Reply:
No, let's not add features in this PR. My comment above was about tc-java configuring the ES JVM specifically to use 2GB.

I am pretty sure the used heap will else depend on the available memory.

What is the intent behind the 3G limit here? To not have the ES container use up all memory of the Docker host?

Reply:
without the 3G specified, it failed to start on my machines (ubuntu 16/32 gig ram). sure, there are some fancy ways containers/apps/jvm determine their memory sizes, but from my experience, it just helps to be explicit.

Reply:
this is one of the outstanding items of #391 but it makes most sense to merge after implementing transferable